templates.go

```diff
@@ -3,27 +3,320 @@ package cloudconfig
 const (
 	MasterTemplate = `#cloud-config
 write_files:
-- path: /srv/10-calico.conf
+- path: /srv/calico-policy-controller-sa.yaml
   owner: root
-  permissions: 0755
+  permissions: 644
   content: |
-    {
-        "name": "calico-k8s-network",
-        "type": "calico",
-        "etcd_endpoints": "https://{{ .Cluster.Etcd.Domain }}:2379",
-        "log_level": "info",
-        "ipam": {
-            "type": "calico-ipam"
-        },
-        "mtu": {{.Cluster.Calico.MTU}},
-        "policy": {
-            "type": "k8s",
-            "k8s_api_root": "https://{{.Cluster.Kubernetes.API.Domain}}/api/v1/",
-            "k8s_client_certificate": "/etc/kubernetes/ssl/calico/client-crt.pem",
-            "k8s_client_key": "/etc/kubernetes/ssl/calico/client-key.pem",
-            "k8s_certificate_authority": "/etc/kubernetes/ssl/calico/client-ca.pem"
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: calico-policy-controller
+      namespace: kube-system
+- path: /srv/calico-node-sa.yaml
+  owner: root
+  permissions: 644
+  content: |
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: calico-node
+      namespace: kube-system
+- path: /srv/calico-configmap.yaml
+  owner: root
+  permissions: 644
+  content: |
+    # Calico Version v2.2.1
+    # http://docs.projectcalico.org/v2.2/releases#v2.2.1
+    # This manifest includes the following component versions:
+    #   calico/node:v1.2.1
+    #   calico/cni:v1.8.3
+    #   calico/kube-policy-controller:v0.6.0
+
+    # This ConfigMap is used to configure a self-hosted Calico installation.
+    kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: calico-config
+      namespace: kube-system
+    data:
+      # Configure this with the location of your etcd cluster.
+      etcd_endpoints: "https://{{ .Cluster.Etcd.Domain }}:2379"
+
+      # Configure the Calico backend to use.
+      calico_backend: "bird"
+    
+      # The CNI network configuration to install on each node.
+      # TODO: Do we still need to set MTU manually?
```

Comment:
@calvix mentioned we need it in KVM, so I created a ticket in operator board for making it optional